client/asset/btc/rpcclient.go

```diff
@@ -136,9 +144,64 @@ func (wc *rpcClient) connect(ctx context.Context, _ *sync.WaitGroup) error {
 	return nil
 }
 
+// reconfigure attempts to reconfigure the rpcClient for the new settings. Live
+// reconfiguration is only attempted if the new wallet type is walletTypeRPC. If
+// the special:activelyUsed flag is set, reconfigure will fail if we can't
+// validate ownership of the current deposit address.
+func (wc *rpcClient) reconfigure(cfg *asset.WalletConfig, currentAddress string) (restartRequired bool, err error) {
+	// rpcClient only handles walletTypeRPC.
+	if cfg.Type != walletTypeRPC {
+		restartRequired = true
+		return
+	}
+
+	parsedCfg := new(RPCWalletConfig)
+	if err = config.Unmapify(cfg.Settings, parsedCfg); err != nil {
+		return
+	}
+
+	// Check the RPC configuration.
+	newCfg := &parsedCfg.RPCConfig
+	if err = dexbtc.CheckRPCConfig(&newCfg.RPCConfig, wc.cloneParams.WalletInfo.Name, wc.cloneParams.Network, wc.cloneParams.Ports); err != nil {
+		return
+	}
+
+	// If the RPC configuration has changed, try to update the client.
+	oldCfg := wc.rpcConfig
+	if newCfg.RPCBind != oldCfg.RPCBind ||
+		newCfg.RPCUser != oldCfg.RPCUser ||
+		newCfg.RPCPass != oldCfg.RPCPass ||
+		newCfg.WalletName != oldCfg.WalletName {
```

Comment:
I think we can directly compare the structs, never have to worry about added fields as long as it's a plain-old-data struct with no pointers. `if *oldCfg == *newCfg { return }`?